common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -136,109 +137,164 @@ void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
 
   uniqID = 1;
 
-  const char* tool_sample = getenv("KOKKOS_TOOLS_SAMPLER_SKIP");
+  const char* tool_sample      = getenv("KOKKOS_TOOLS_SAMPLER_SKIP");
+  const char* tool_probability = getenv("KOKKOS_TOOLS_SAMPLER_PROBABILITY");
+  // composing the periodicity and probability parameter. Set to 1 if
+  // probability for each periodic sample. Set to 0 to default to probability
+  // even if periodicity (skip rate) is defined.
+
   if (NULL != tool_sample) {
     kernelSampleSkip = atoi(tool_sample) + 1;
   }
 
+  if (NULL != tool_probability) {
+    //  read sampling probability as an integer between 1 and 100, but
+    //  programs reasons about probability as a double between 0.0 and 1.0.
+    tool_prob_num = atoi(tool_probability);
```

Comment:
we should maybe allow something less than 1% can we read in floating point numbers?

Reply:
maybe say tool probability is in % so values are between 0 and 100 but we allow you to set the env variable to 0.1

Reply:
0.1 would mean 1 in a 1000